### ISSUE TITLE
Simple order cycle edit page loads slowly with many products

### ISSUE THREAD

**Issue by mkllnk** [openfoodfoundation/openfoodnetwork#4755]
### What we should change and why (this is tech debt)

We are loading all visible products into the simple order cycle page. This doesn't perform well with thousands of products and a user may think the page is broken or we hit timeouts. We should introduce pagination similar to the more complex order cycle edit page.

### Context

After introducing pagination for the complex order cycle page, the simple page was loading only the first 100 products (issue #4754). We implemented a quick fix to load all products again (#4744). Implementing pagination would have taken a lot longer. A first spike has been done in #4743 but needs more work:

-   add summary count "25 of 130 Variants loaded"
-   we would need to make the "select all" button work with paginated results
-   we would need to make the simple create OC page work correctly as well
-   remove copy pasted code - the code should not be repeated in two angular controllers

### Impact and timeline

This could be a problem for the German instance. The performance problem hasn't been reported in production yet.

**Comment by luisramos0** [comment]
I dont think this issue is tech-debt, it's performance, it's a different label. It basically means it can be prioritized by product. In this case, I think "paginate simple OC edit page" can be a product request at this point.
